src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -247,11 +247,47 @@ public AzureAppConfigurationOptions Connect(Uri endpoint, TokenCredential creden
             }
 
             ConnectionString = null;
-            Endpoint = endpoint;
+            Endpoints = new List<Uri>() { endpoint };
             Credential = credential;
             return this;
         }
 
+        /// <summary>
+        /// Connect the provider to Azure App Configuration and their replicas using list of endpoints and token credential.
+        /// </summary>
+        /// <param name="endpoints">The list of endpoints of the Azure App Configuration and it's replicas to connect to.</param>
+        /// <param name="credential">Token credential to use to connect.</param>
+        public AzureAppConfigurationOptions Connect(IEnumerable<Uri> endpoints, TokenCredential credential)
+        {
+            if (endpoints == null || endpoints.Count() == 0)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            var set = new HashSet<String>();
+
+            // Best effort to find the primary endpoint (the shortest one in length)
+            var shortestEndpoint = endpoints.OrderBy(e => e.Host.Length).First().Host;
```

Comment:
I don't see the need to find the shortest. The first endpoint in the list will do the same job.